MonoGame.Framework/Graphics/Effect/EffectParameterCollection.cs

```diff
@@ -42,7 +42,7 @@ public int Count
                 // TODO: Add a name to parameter lookup table.
 				foreach (var parameter in _parameters) 
                 {
-					if (parameter.Name == name) 
+					if (parameter.Name == name || parameter.Name == name+"Sampler") 

```

Comment:
This i am fairly sure is not correct behavior or at least something that should be fixed in the shader generation and not at runtime.  I would want to see some unit test to verify this.